spec/electra/singleattestation_json.go

```diff
@@ -0,0 +1,80 @@
+// Copyright © 2024 Attestant Limited.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package electra
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/pkg/errors"
+)
+
+// singleAttestationJSON is the spec representation of the struct.
+type singleAttestationJSON struct {
+	CommitteeIndex string                  `json:"committee_index"`
+	AttesterIndex  string                  `json:"attester_index"`
+	Data           *phase0.AttestationData `json:"data"`
+	Signature      string                  `json:"signature"`
+}
+
+// MarshalJSON implements json.Marshaler.
+func (a *SingleAttestation) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&singleAttestationJSON{
+		CommitteeIndex: fmt.Sprintf("%d", a.CommitteeIndex),
+		AttesterIndex:  fmt.Sprintf("%d", a.AttesterIndex),
+		Data:           a.Data,
+		Signature:      fmt.Sprintf("%#x", a.Signature),
+	})
+}
+
+// UnmarshalJSON implements json.Unmarshaler.
+func (a *SingleAttestation) UnmarshalJSON(input []byte) error {
+	var singleAttestationJSON singleAttestationJSON
+	err := json.Unmarshal(input, &singleAttestationJSON)
+	if err != nil {
+		return errors.Wrap(err, "invalid JSON")
+	}
+
+	return a.unpack(&singleAttestationJSON)
+}
+
+func (a *SingleAttestation) unpack(singleAttestationJSON *singleAttestationJSON) error {
+	var err error
+	if singleAttestationJSON.CommitteeIndex == "" {
```

Comment:
It doesn't look like committee index or attester index is actually populated.  We should have at the least a test for a correct JSON version of the attestation, and ideally the standard incorrect ones as well.  The phase0 tests for attestation can be used as a basis.